differs/rpm_diff.go

```diff
@@ -87,7 +95,65 @@ func (a RPMAnalyzer) getPackages(image pkgutil.Image) (map[string]util.PackageIn
 		}
 	}
 
-	return rpmDataFromContainer(image)
+	packages, err := rpmDataFromImageFS(image)
+	if err != nil {
+		logrus.Warn("Trying to run the RPM binary of the image in a container")
+		return rpmDataFromContainer(image)
+	}
+	return packages, err
+}
+
+// rpmDataFromImageFS runs a local rpm binary, if any, to query the image
+// rpmdb and returns a map of installed packages.
+func rpmDataFromImageFS(image pkgutil.Image) (map[string]util.PackageInfo, error) {
+	packages := make(map[string]util.PackageInfo)
+	// Check there is an executable rpm tool in host
+	if err := exec.Command("rpm", "--version").Run(); err != nil {
+		logrus.Warn("No RPM binary in host")
+		return packages, err
+	}
+	dbPath, err := rpmDBPath(image.FSPath)
+	if err != nil {
+		logrus.Warnf("Couldn't find RPM database: %s", err.Error())
+		return packages, err
+	}
+	cmdArgs := append([]string{"--root", image.FSPath, "--dbpath", dbPath}, rpmCmd[1:]...)
+	out, err := exec.Command(rpmCmd[0], cmdArgs...).Output()
+	if err != nil {
+		logrus.Warnf("RPM call failed: %s", err.Error())
+		return packages, err
+	}
+	output := strings.Split(string(out), "\n")
+	return parsePackageData(output)
+}
+
+// rpmDBPath tries to get the RPM database path from the /usr/lib/rpm/macros
+// file in the image rootfs.
+func rpmDBPath(rootFSPath string) (string, error) {
+	rpmMacros, err := os.Open(filepath.Join(rootFSPath, "usr/lib/rpm/macros"))
+	if err != nil {
+		return "", err
+	}
+	defer rpmMacros.Close()
+
+	scanner := bufio.NewScanner(rpmMacros)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(line, "%_dbpath") {
+			fields := strings.Fields(line)
+			if len(fields) < 2 {
```

Comment:
could you add a comment with an example of what you're trying to match for here? maybe a regex would be more clear?

Reply:
I will add a comment with the example, I think this is the clearest.